core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -170,6 +170,13 @@ class KafkaApis(val requestChannel: RequestChannel,
     val response =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         val (result, error) = replicaManager.stopReplicas(stopReplicaRequest)
+        result.map{case (topicPartition, errorCode) => {
```

Comment:
If a broker is already a follower for a __consumer_offsets partition, wouldn't the cache already been empty? If the broker was never a leader, the cache is obviously empty. If it used to be a leader but became a follower later, it must have received a leader and ISR and cleared its cache. In any case a follower broker's cache should already be empty, right? 

Looking at the current code, there seems some issues if a coordinator transition happened during consumer rebalance. For example, if the following sequence happens.
0. The broker is initially a leader
1. A JoinGroupRequest comes from consumer 0 and passed the leader check
2. The broker becomes a follower
3. The JoinGroupRequest in step 2 added the group to the cache.
4. The JoinGroupRequest in step 2 returns later on and consumer 0 will send a SyncGroupRequest to the old leader and get an error. The cache won't be cleaned up on the follower in this case. 
 
Depending on the timing of when the leader change happens, if another consumer tries to join the group at the same time, it is possible that two consumers will consume the same partition as you saw. But that might be expected and one of the consumer should not be able to commit offset.

So it looks the key issue here is that we have to somehow synchronize the group state change with the coordinator movement to prevent a stale coordinator from returning a non-error response. For the stale cache case you mentioned specifically, when an Not_Coordinator_For_Group error is returned, we should clean up the group information.

@hachikuji It would be good if you can also take a look and see if that is a problem. Thanks.